hazelcast/src/main/java/com/hazelcast/map/impl/client/MapRemoveRequest.java

```diff
@@ -80,7 +79,7 @@ protected void beforeResponse() {
     }
 
     protected Operation prepareOperation() {
-        RemoveOperation op = new RemoveOperation(name, key);
+        MapOperation op = getOperationProvider().createRemoveOperation(name, key);
```

Comment:
It may be but i preserved current parameter-list in constructors. It may be a subject of another PR later.